replacement_mode_modeling/01_extract_db_data.ipynb

```diff
@@ -0,0 +1,1645 @@
+   "id": "a6c20466",
+   "metadata": {},
+   "source": [
+    "### Outlier removal"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c05071cc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"For {CURRENT_DB=}, before outlier removal, n_rows = {filtered_trips.shape[0]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b222715f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Drop instances where duration/distance is unusable.\n",
+    "filtered_trips.drop(\n",
+    "    index=filtered_trips.loc[(filtered_trips.section_distance_argmax <= 0) | (filtered_trips.section_duration_argmax <= 0), :].index,\n",
+    "    inplace=False\n",
+    ").reset_index(drop=True, inplace=True)\n",
+    "\n",
+    "\n",
+    "# bus, train, bicycling, walking, car\n",
+    "# split-apply-combine\n",
+    "def drop_outliers(df: pd.DataFrame, low=0.1, high=0.9) -> pd.DataFrame:\n",
+    "    \n",
+    "    def filter_by_percentiles(group):\n",
+    "        distance_low = group['section_distance_argmax'].quantile(low)\n",
+    "        distance_high = group['section_distance_argmax'].quantile(high)\n",
+    "        duration_low = group['section_duration_argmax'].quantile(low)\n",
+    "        duration_high = group['section_duration_argmax'].quantile(high)\n",
+    "        \n",
+    "        l1_filter = group[\n",
+    "            (group['section_distance_argmax'] >= distance_low) &\n",
+    "            (group['section_distance_argmax'] <= distance_high)\n",
+    "        ].reset_index(drop=True)\n",
+    "        \n",
+    "        l2_filter = l1_filter[\n",
+    "            (l1_filter['section_duration_argmax'] >= duration_low) &\n",
+    "            (l1_filter['section_duration_argmax'] <= duration_high)\n",
+    "        ].reset_index(drop=True)\n",
+    "        \n",
+    "        return l2_filter\n",
+    "    \n",
+    "    return df.groupby('section_mode_argmax').apply(filter_by_percentiles).reset_index(drop=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d77febb3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_trips = drop_outliers(filtered_trips, low=0.01, high=0.99)\n",
+    "\n",
+    "# Ideal speed. distance/time (in hours).\n",
+    "filtered_trips['mph'] = (\n",
+    "    (filtered_trips['section_distance_argmax'] * 60.)/filtered_trips['section_duration_argmax']\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b52d5325",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_trips[['section_mode_argmax', 'section_duration_argmax', 'section_distance_argmax', 'mph']].head(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c7ed953d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def filter_mph(df: pd.DataFrame, low=0.1, high=0.9) -> pd.DataFrame:\n",
```

Comment:
Yes, it removes trips that fall outside given range of percentiles. The default arguments are configured to be the 10th and 90th percentile, but in the subsequent cell, I call the method by specifying the 1st and 99th percentile as the outlier ranges. It is fairly standard preprocessing practice to remove these outliers, especially because such outliers can easily dominate/influence any aggregation-based analyses.

Reply:
So they're removed based on their position as outliers (by percentile) not by these hard cutoff values? I was looking at the 4mph walking and 15mph biking cutoffs

Reply:
Following up on this based on our call today - it seems that yes, we are cutting off sections over these levels right now, but plan to re-evaluate this decision in the future -- maybe remove outliers since we have data to determine what those are rather than using the outside source.